openjdk.test.load/src/test.load/net/adoptopenjdk/stf/MixedLoadTest.java

```diff
@@ -63,16 +80,37 @@ public void execute(StfCoreExtension test) throws StfException {
 				.addProjectToClasspath("openjdk.test.nio")
 				.addProjectToClasspath("openjdk.test.concurrent")
 				.addJarToClasspath(mauveJar)
-				.addJarToClasspath(fileSystemJar)
-				.setAbortIfOutOfMemory(false)
-				.addSuite("mix")
-				.setSuiteThreadCount(cpuCount - 2, 10)	
-				.setSuiteNumTests(totalTests * 200) // Leave 1 cpu for the JIT. i for GC and set min 20			
-				.setSuiteInventory(inventoryFile) 			// Point at the file which lists the tests
-				.setSuiteRandomSelection();		  			// Randomly pick the next test each time
+				.addJarToClasspath(fileSystemJar);
 				
+		if (isTimeBasedLoadTest) { 
+			loadTestInvocation = loadTestInvocation.setTimeLimit(timeLimit);	// If it's a time based test, stop execution after given time duration
+		}
+		loadTestInvocation = loadTestInvocation.setAbortIfOutOfMemory(false)
+				.addSuite("mix")
+				.setSuiteThreadCount(cpuCount - 2, 10); // Leave 1 cpu for the JIT, 1 cpu for GC and set min 10
+		if (isTimeBasedLoadTest) { 
+			loadTestInvocation = loadTestInvocation
+				.setSuiteNumTests(totalTests * 80000000 * testCountMultiplier); // If it's a time based test, run a very large load
```

Comment:
@Mesbah-Alam shouldn't we just set this to `INT_MAX` or `LONG_MAX` or whatever the maximum representable integer is rather than using a "very large load" with a magic 80,000,000 constant? The tests will be terminated after the alloted time anyway.

Reply:
Thanks for your suggestion. I will try this out.